### ISSUE TITLE
Split visualize.storm into subfunctions

### ISSUE THREAD

**Issue by aappling-usgs** [USGS-VIZLAB/vizstorm#4]
(where visualize.storm starts as something like https://github.com/USGS-VIZLAB/hurricane-irma/blob/master/scripts/visualize/visualize.hurricane_map.R)

Decompose visualize.storm into more subfunctions for things like base map with info, gage points, sparklines, legend, watermark

**Comment by aappling-usgs** [comment]
What's the difference between the svg_base_map and hurricane_map? It's not that one is static while the other is animated...links to #10

**Comment by jordansread** [comment]
I think the idea with `base_map` is it is used to get all of the geometries into the svg in the right place and with the right ids

The second function was used to add sparklines, legend, and in some cases, copy elements and make them transparent (so that their mouse events are on "top" but their visual appears below)

**Comment by aappling-usgs** [comment]
Which things count as geometries? I ask because sparklines and legend circles and even legend text sound to me like they could be geometries.

**Comment by jordansread** [comment]
I would agree with that. There is a potential divider between geo (map) elements that are placed according to their coordinates and the projection, and then other geometries that are created and placed according to the svg bounds (such as, place this watermark 10px from the upper right). I think legends and sparks fall into the latter category.

**Comment by aappling-usgs** [comment]
OK, thanks. I agree that it makes sense to pick some rule for dividing the functions, and geographic elements vs annotation elements seems sensible, too. Let's just get some more text about that into comments as we harden this.
